### DISCUSSION TITLE
best practises for extend a spacy language model

### DISCUSSION THREAD

**Discussion by joseberlines** [explosion/spaCy#12274] [Help: Best practices]
I work with several of the languages models available in spacy.
I have some functionalities as follows:

1. tweak the tokenizer
2. several pipelines that add atrributes to tokens (extensions)
3. several pipelines that add extensions to certain spans (based on regular expressions)

What is the bes practise to create a spacy doc object that takes an existing language model, applies all the changes and gives back a spacy object with all the extra functionalities.

MOreover, is that same procedure when trying to extend language models of the so called spacy universe (medspacy for instance)? (for instance scispacy?

Thanks.

**Comment by kadarakos** [comment]
Hey,

To be able to provide a useful answer may I ask for some more clarification? Perhaps a concrete example to start the discussion from?
